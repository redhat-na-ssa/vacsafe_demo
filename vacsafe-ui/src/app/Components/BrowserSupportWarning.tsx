import React from 'react';
import { Alert } from '@patternfly/react-core';
import UAParser from 'ua-parser-js';
const BROWSER_SUPPORT = ['Firefox', 'Chrome', 'Mobile Safari'];
const OS_SUPPORT = ['Linux', 'Mac OS', 'Windows', 'Android', 'iOS']
const parser = new UAParser();
const result = parser.getResult();

const browserCompatibility = (result: { os: { name: string; }; browser: { name: string; }; }) => {
  if (!OS_SUPPORT.includes(result.os.name)) {
    return messageRender(result.os.name);
  }
  if (!BROWSER_SUPPORT.includes(result.browser.name)) {
    return messageRender(result.browser.name);
  }
  return null;
}

const messageRender = (value : string) => (
  <Alert isInline variant="info" title="Browser Warning">
      <p>
      We noticed you are using{' '}
      <span>
       <strong>
         {value ? value : "an unsupported browser"}
       </strong>, which we do not currently support
     </span>. You may experience issues.
      </p>
      <p>
        Recommended Browsers:{' '}
          <a 
            href="https://www.google.com/chrome/"
            rel="noreferrer"
            target="_blank"
          >
            Chrome
          </a>,{' '}
          <a
            href="https://www.mozilla.org/en-US/firefox/new/"
            rel="noreferrer"
            target="_blank"
          >
            Firefox
          </a>,{' '}
          <span>
            Safari on iOS
          </span>
      </p>
    </Alert>
)

const BrowserSupportWarning: React.FunctionComponent = () => (browserCompatibility(result));

export default BrowserSupportWarning;
