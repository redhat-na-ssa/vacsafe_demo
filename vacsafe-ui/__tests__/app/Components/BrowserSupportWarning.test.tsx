import React from 'react';
import { shallow } from 'enzyme';
import BrowserSupportWarning from '../../../src/app/Components/BrowserSupportWarning';

describe('<BrowserSupportWarning />', ()=> {
    it('should be defined', () => {
        expect(BrowserSupportWarning).toBeDefined()
    })
    it("should render without crashing", () => {
        const wrapper = shallow(<BrowserSupportWarning />)
        expect(wrapper)
    })
    it("should match existing snapshot", () => {
        const wrapper = shallow(<BrowserSupportWarning />)
        expect(wrapper).toMatchSnapshot()
    })
})
