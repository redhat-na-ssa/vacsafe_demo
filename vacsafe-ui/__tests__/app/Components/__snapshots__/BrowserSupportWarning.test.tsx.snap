// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrowserSupportWarning /> should match existing snapshot 1`] = `
<Alert
  isInline={true}
  title="Browser Warning"
  variant="info"
>
  <p>
    We noticed you are using
     
    <span>
      <strong>
        an unsupported browser
      </strong>
      , which we do not currently support
    </span>
    . You may experience issues.
  </p>
  <p>
    Recommended Browsers:
     
    <a
      href="https://www.google.com/chrome/"
      rel="noreferrer"
      target="_blank"
    >
      Chrome
    </a>
    ,
     
    <a
      href="https://www.mozilla.org/en-US/firefox/new/"
      rel="noreferrer"
      target="_blank"
    >
      Firefox
    </a>
    ,
     
    <span>
      Safari on iOS
    </span>
  </p>
</Alert>
`;
